### PR TITLE
Feat: adds ios and android push notifications

### PR DESCRIPTION
**Pull request by vitorguima**

### CODE REVIEW

File: baseapp-notifications/baseapp_notifications/tasks.py [hunk 1]
```diff
@@ -1,11 +1,29 @@
...
+
+    # send messages to all Windows devices
+    windows_devices = WNSDevice.objects.filter(user__id=user_id)
+    windows_devices.send_message(**message_data)
+
+    # send messages to all Web devices
+    web_devices = WebPushDevice.objects.filter(user__id=user_id)
+    web_devices.send_message(**message_data)
+
     pass
```
Comment (nossila) [inline baseapp-notifications/baseapp_notifications/tasks.py:29]: You can remove `pass` here

File: baseapp-notifications/setup.cfg
Comment (nossila) [inline baseapp-notifications/setup.cfg:1]: Increase the `version` to `0.1.4`

**Review by nossila — CHANGES_REQUESTED** [review]
Would be nice to add a section to `README` about how to setup push notifications.

File: baseapp-notifications/baseapp_notifications/utils.py [hunk 2]
```diff
@@ -72,15 +75,13 @@ def send_notification(
             notification.save(update_fields=["emailed"])
 
     if send_push:
-        send_push_notification.delay(
+        kwargs.pop("level", None)
+        send_push_notification(
```
Comment (nossila) [inline baseapp-notifications/baseapp_notifications/utils.py:79]: revert this
```suggestion
        send_push_notification.delay(
```

File: baseapp-notifications/baseapp_notifications/utils.py [hunk 3]
```diff
@@ -72,15 +75,13 @@ def send_notification(
             notification.save(update_fields=["emailed"])
 
     if send_push:
-        send_push_notification.delay(
+        kwargs.pop("level", None)
```
Comment (nossila) [inline baseapp-notifications/baseapp_notifications/utils.py:78]: maybe we can do this pop inside the `send_push_notification` function?

File: baseapp-notifications/baseapp_notifications/utils.py [hunk 4]
```diff
@@ -72,15 +75,13 @@ def send_notification(
...
+        send_push_notification(
             recipient.id,
-            dict(
-                description=description,
-                verb=verb,
-                **kwargs
-                # TO DO:
-                # serialize all objects so devices can use this data if necessary
-            ),
+            push_title=push_title,
```
Comment (nossila) [inline baseapp-notifications/baseapp_notifications/utils.py:81]: I think `push_title` could be something like this:

```suggestion
            push_title=push_title or settings.MOBILE_APP_NAME,
```

I noticed this as pattern, when a title is not set it uses the APP_NAME... or iOS and Android already handle this when we sent an empty or None title?

**Review by nossila — CHANGES_REQUESTED** [review]

File: baseapp-notifications/baseapp_notifications/tasks.py [hunk 5]
```diff
@@ -1,11 +1,34 @@
...
-def send_push_notification(user_id, data):
-    # TO BE DONE
-    # get user's device tokens and send to each of them
-    # this function should be easy to override per project since
-    # some projects might use different push notification services
-    # like, expo, firebase, APN and so on
-    pass
+def send_push_notification(user_id, **kwargs):
+    kwargs.pop("level", None)
+    push_title = kwargs.pop("push_title", None)
```
Comment (vitorguima) [inline baseapp-notifications/baseapp_notifications/tasks.py:7]: move it to definition params

**Review by nossila — APPROVED** [review]

**Review by nossila — APPROVED** [review]

**Pull request merged**